core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/utilities/DummyTextAnnotationGenerator.java

```diff
@@ -195,113 +212,151 @@ public static TextAnnotation generateBasicTextAnnotation(int numSentences) {
+                        lemmaView.addTokenLabel(i, lemmaOveral1[i], 1.0);
+                    if (sentenceLength > 1)
+                        for (int i = 0; i < lemmaOveral2.length; i++)
+                            lemmaView.addTokenLabel(lemmaOveral1.length + i, lemmaOveral2[i], 1.0);
+                    if (sentenceLength > 2)
+                        for (int i = 0; i < lemmaOveral3.length; i++)
+                            lemmaView.addTokenLabel(lemmaOveral1.length + lemmaOveral2.length +  i, lemmaOveral3[i], 1.0);
                     ta.addView(viewName, lemmaView);
                     break;
                 case ViewNames.SHALLOW_PARSE:
                     SpanLabelView chunkView = new SpanLabelView(ViewNames.SHALLOW_PARSE, ta);
-                    if (withNoisyLabels)
-                        for (IntPair span : chunks_noisy.keySet())
+                    Map<IntPair, String> chunkOverall1 =
+                            (withNoisyLabels) ? chunks_noisy1 : chunks1;
+                    Map<IntPair, String> chunkOverall2 =
+                            (withNoisyLabels) ? chunks_noisy2 : chunks2;
+                    Map<IntPair, String> chunkOverall3 =
+                            (withNoisyLabels) ? chunks_noisy3 : chunks3;
+                    for (IntPair span : chunkOverall1.keySet())
+                        chunkView.addSpanLabel(span.getFirst(), span.getSecond(),
+                                chunkOverall1.get(span), 1.0);
+                    if (sentenceLength > 1)
+                        for (IntPair span : chunkOverall2.keySet())
                             chunkView.addSpanLabel(span.getFirst(), span.getSecond(),
-                                    chunks_noisy.get(span), 1.0);
-                    else
-                        for (IntPair span : chunks.keySet())
+                                    chunkOverall2.get(span), 1.0);
+                    if (sentenceLength > 2)
+                        for (IntPair span : chunkOverall3.keySet())
                             chunkView.addSpanLabel(span.getFirst(), span.getSecond(),
-                                    chunks.get(span), 1.0);
+                                    chunkOverall3.get(span), 1.0);
                     ta.addView(ViewNames.SHALLOW_PARSE, chunkView);
                     break;
                 case ViewNames.NER_CONLL:
-                    SpanLabelView nerView = new SpanLabelView(ViewNames.NER_CONLL, ta );
-                    Map<IntPair, String> nerSource = (withNoisyLabels ? ner_noisy : ner );
-                    for (IntPair span : nerSource.keySet() )
-                        nerView.addSpanLabel(span.getFirst(), span.getSecond(), ner_noisy.get( span ), 1.0 );
-                    ta.addView( ViewNames.NER_CONLL, nerView );
+                    SpanLabelView nerView = new SpanLabelView(ViewNames.NER_CONLL, ta);
+                    Map<IntPair, String> nerSource1 = withNoisyLabels ? ner_noisy1 : ner1;
```

Comment:
No () on the conditional `withNoisyLabels`? I didn't know you can do that (also, not consistent with previous checks.